### PR TITLE
Added type hinting

### PR DESCRIPTION
**Pull request by 0xrameshh**
# Type hinting for all the files

Hello, I have added type hinting for all the files in the repo. hope this helps.

### CODE REVIEW

File: docker-stubs/api/build.pyi [hunk 1]
```diff
@@ -1,38 +1,32 @@
-from .. import auth as auth, constants as constants, errors as errors, utils as utils
-from _typeshed import Incomplete
+from typing import Optional, Union
+from .. import errors as errors, utils as utils
+from ..types import CancellableStream as CancellableStream
 
-log: Incomplete
-
-class BuildApiMixin:
```
Comment (vdozier-work) [inline docker-stubs/api/build.pyi:6]: why is the class name changed here?

File: docker-stubs/api/container.pyi [hunk 2]
```diff
@@ -9,145 +8,131 @@ from ..types import (
...
+        container: str,
+        repository: Optional[str] = None,
+        tag: Optional[str] = None,
+        message: Optional[str] = None,
+        author: Optional[str] = None,
+        pause: bool = True,
+        changes: Optional[list[str]] = None,
+        conf: Optional[ContainerConfig] = None,
+    ) -> dict[str, str]: ...
     def containers(
```
Comment (vdozier-work) [inline docker-stubs/api/container.pyi:41]: do we really want to include the default arguments here? I'm worried they'll fall out of step with the actual values in the library

Comment (vdozier-work) [reply docker-stubs/api/container.pyi:41]: I believe there is a `= ...` syntax option now to show in a stub that there are default arguments

Comment (Tatsh) [reply docker-stubs/api/container.pyi:41]: Why not use the new optional syntax? And yes you are not supposed to put the actual for the default but `...`.

Comment (vdozier-work) [reply docker-stubs/api/container.pyi:41]: there's new optional syntax?

**Pull request closed without merging**